scripts/checkout_submodules.py

```diff
@@ -101,9 +105,16 @@ def checkout_modules(modules: list, shallow: bool, force: bool, recursive: bool,
     cmd += ['--force'] if force else []
     cmd += ['--recursive'] if recursive else []
     cmd += ['--jobs', f'{jobs}'] if jobs else []
-    cmd += [module.path for module in modules]
+    module_paths = [module.path for module in modules]
 
-    subprocess.check_call(cmd)
+    subprocess.check_call(cmd + module_paths)
+
+    if not recursive:
```

Comment:
could we split out this change separately? I would like to review it and merge it as a stand-alone.

My first instinct reading just this like is "why do we have if not recursive add recursive" .... seems somehow reversed.

Reply:
I split this change out to https://github.com/project-chip/connectedhomeip/pull/33089 and changed some expressions to make it easier to read.